### PR TITLE
Update Place freshness logic

### PR DESCRIPTION
**Pull request by timwangmusic**
## Description
Previously, we determine if a new API call to the Maps is necessary when the number of results is lower than requested. However, with the new Maps query fixed on min/max price level, this will not work properly since higher priced restaurants are inevitably less than the requested most of the time. Therefore the new logic says as long as we have some data for the location with the requested category and price level, it is acceptable data from the database.

## Solution
* `MapsSearchTime` entries have `PriceLevel` info now
* Update place freshness if statement

## Testing
- [ ] Integration testing on Heroku staging
- [x] Added new unit tests

## Checks
- [x] Have you removed commented code?
- [x] Have you used gofmt to format your code?

### CODE REVIEW

File: iowrappers/poi_searcher.go [hunk 1]
```diff
@@ -144,17 +144,21 @@ func (s *PoiSearcher) NearbySearch(context context.Context, request *PlaceSearch
...
 	// update last search time for the city
-	lastSearchTime, cacheMiss := s.redisClient.GetMapsLastSearchTime(context, location, request.PlaceCat)
+	lastSearchTime, lastSearchTimeMiss := s.redisClient.GetMapsLastSearchTime(context, location, request.PlaceCat, request.PriceLevel)
 
 	currentTime := time.Now()
-	// use place data from database if the location is known and the data is fresh, and we have sufficient data
-	if cacheMiss == nil && (currentTime.Sub(lastSearchTime) <= MinMapsResultRefreshDuration && uint(len(cachedPlaces)) >= request.MinNumResults) {
-		Logger.Infof("[request_id: %s] Using Redis to fulfill request. Place Type: %s", context.Value(ContextRequestIdKey), request.PlaceCat)
+	// use place data from the database if the data is fresh, and we have at least one known place satisfying the request
+	if currentTime.Sub(lastSearchTime) <= MinMapsResultRefreshDuration && lastSearchTimeMiss == nil && len(cachedPlaces) > 0 {
```
Comment (rwangsc18) [inline iowrappers/poi_searcher.go:151]: Should we first check (lastSearchTimeMiss == nil) here, in case lastSearchTime returns some invalid result?

Comment (rwangsc18) [reply iowrappers/poi_searcher.go:151]: nit: we can also turn IF condition into a utility function? say `IsCacheResultFresh()`. Then we can make this line shorter and even remove the long comment.

**Review by rwangsc18 — APPROVED** [review]
Overall LGTM. left one comment.

File: iowrappers/poi_searcher.go (hunk 1 above)
Comment (timwangmusic) [reply iowrappers/poi_searcher.go:151]: Good point of making the IF condition as a utility function. Will do. The reason I am moving the `lastSearchTimeMiss == nil` to the second place is freshness is the top priority regardless of cache miss or not.

**Pull request merged**